### DISCUSSION TITLE
How do I automatically run tmux on startup instead of typing it everytime I open a new tab?

### DISCUSSION THREAD

**Discussion by allexus13** [GitSquared/edex-ui#1143] [Questions]
I just discovered this terminal emulator and it was fascinating but I wonder if there's a file that can handle **startup commands** like **.bash_profile**? I want to run **tmux** on startup. It works fine when I manually activate it on a tab but could be tedious when I work with many tabs. settings.json and shortcuts.json can't seem to do it or can it?

**Comment by GitSquared** [comment]
You can set the shellArgs option to something like `-c tmux` so eDEX launches `bash -c tmux` on startup, for instance. See https://github.com/GitSquared/edex-ui/wiki/settings.json

**Reply by allexus13** [reply]
I believe I tried that but it threw me an error. I will post the error message once I get the chance. Fantastic project by the way. Thanks.

**Reply by allexus13** [reply]
> You can set the shellArgs option to something like `-c tmux` so eDEX launches `bash -c tmux` on startup, for instance. See https://github.com/GitSquared/edex-ui/wiki/settings.json

so I tried:
-c tmux
'-c tmux'
"-c tmux"
tmux
'tmux'
"tmux"

![image](https://user-images.githubusercontent.com/50406751/126778445-20efe84a-9f2a-415d-aacb-c4f7fbe023e7.png)

the error says "string is not support as args in unix"

![image](https://user-images.githubusercontent.com/50406751/126778614-5cccbae9-005b-45d7-86b3-624fc28e065e.png)

Am I doing something wrong or missing something? I'm still trynna play around with the settings and stuff.

**Reply by allexus13** [reply]
> > You can set the shellArgs option to something like `-c tmux` so eDEX launches `bash -c tmux` on startup, for instance. See https://github.com/GitSquared/edex-ui/wiki/settings.json
> 
> so I tried:
> -c tmux
> '-c tmux'
> "-c tmux"
> tmux
> 'tmux'
> "tmux"
> 
> ![image](https://user-images.githubusercontent.com/50406751/126778445-20efe84a-9f2a-415d-aacb-c4f7fbe023e7.png)
> 
> the error says "string is not support as args in unix"
> 
> ![image](https://user-images.githubusercontent.com/50406751/126778614-5cccbae9-005b-45d7-86b3-624fc28e065e.png)
> 
> Am I doing something wrong or missing something? I'm still trynna play around with the settings and stuff.

other values such as:
-c whoami
does not work and produce the same error. If there's anything you want me to try or to help fix this, just tell me.

**Reply by GitSquared** [reply]
> string is not support as args in unix

Sounds like an issue with node-pty not supporting the way we send shell params, will investigate - thanks for the logs

**Reply by ghost** [reply]
node-pty takes an array of strings rather than a single string.  If you set shellArgs to `['-c','tmux']`in an external editor, it should work.
